### PR TITLE
Google calendar

### PR DESCRIPTION
**Pull request by polyvjk**
- Choose which calendars to sync
  - from new activity after sign in
  - from settings fragment

- Update dirty events in receiver
- Save calendars settings (category) in player

### CODE REVIEW

File: app/src/main/java/io/ipoli/android/app/ui/dialogs/AndroidCalendarsPickerFragment.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        View view = inflater.inflate(R.layout.fragment_calendar_picker, null);
+        unbinder = ButterKnife.bind(this, view);
+
+        List<AndroidCalendarViewModel> viewModels = new ArrayList<>();
+
+        //test on emulator
+//        viewModels.add(new AndroidCalendarViewModel(1L, "Polina Zhelyazkova", Category.PERSONAL, true));
+//        viewModels.add(new AndroidCalendarViewModel(2L, "Vihar calendar", Category.PERSONAL, false));
+//        viewModels.add(new AndroidCalendarViewModel(3L, "Holidays", Category.PERSONAL, false));
+//        viewModels.add(new AndroidCalendarViewModel(4L, "Birthdays", Category.PERSONAL, false));
```
Comment (curiousily) [inline app/src/main/java/io/ipoli/android/app/ui/dialogs/AndroidCalendarsPickerFragment.java]: Obvious?

File: app/src/main/java/io/ipoli/android/app/AndroidCalendarEventParser.java [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+        if (startDate.isBefore(LocalDate.now())) {
+            return null;
+        }
+
+        Quest q = new Quest(event.title);
+        q.setSource(Constants.SOURCE_ANDROID_CALENDAR);
+        q.setSourceMapping(SourceMapping.fromGoogleCalendar(event.calendarId, event.id));
+        q.setCategoryType(category);
+
+        LocalDate endDate = event.dTend > 0 ? DateUtils.toStartOfDayUTCLocalDate(DateUtils.fromMillis(event.dTend)) : startDate;
```
Comment (polyvjk) [inline app/src/main/java/io/ipoli/android/app/AndroidCalendarEventParser.java]: Yes, `DateUtils.fromMillis()` with `ZoneId` as parameter works great

File: app/src/main/java/io/ipoli/android/app/AndroidCalendarEventParser.java [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        Recurrence recurrence = Recurrence.create();
+        recurrence.setFlexibleCount(0);
+        LocalDate startDate = DateUtils.toStartOfDayUTCLocalDate(startLocalDateTime.toLocalDate());
+        recurrence.setDtstartDate(startDate);
+        LocalDate endDate = null;
+        if(event.dTend > 0) {
+            endDate = DateUtils.toStartOfDayUTCLocalDate(DateUtils.fromMillis(event.dTend));
+        } else if(recur.getUntil() != null) {
+            LocalDateTime endLocalDateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(recur.getUntil().getTime()), getZoneId(event));
+            endDate = DateUtils.toStartOfDayUTCLocalDate(endLocalDateTime.toLocalDate());
```
Comment (polyvjk) [inline app/src/main/java/io/ipoli/android/app/AndroidCalendarEventParser.java]: Yes

File: app/src/main/java/io/ipoli/android/app/AndroidCalendarEventParser.java [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+            if (StringUtils.isEmpty(timeZone)) {
+                timeZone = e.calendarTimeZone;
+            }
+        }
+
+        ZoneId zoneId;
+        try {
+            zoneId = StringUtils.isEmpty(timeZone) ? ZoneId.systemDefault() : ZoneId.of(timeZone);
+        } catch (Exception ex) {
+            zoneId = ZoneId.of(e.calendarTimeZone);
```
Comment (polyvjk) [inline app/src/main/java/io/ipoli/android/app/AndroidCalendarEventParser.java]: 👍

File: app/src/main/java/io/ipoli/android/app/App.java [hunk 5]
```diff
@@ -273,7 +275,7 @@ private void listenForDailyQuestsChange() {
             try {
                 localStorage.saveString(Constants.KEY_WIDGET_AGENDA_QUEST_LIST, objectMapper.writeValueAsString(quests));
             } catch (JsonProcessingException e) {
-                throw new RuntimeException("Cant save quests for Widget", e);
+                throw new RuntimeException("Cant saveSync quests for Widget", e);
```
Comment (polyvjk) [inline app/src/main/java/io/ipoli/android/app/App.java]: 👍

File: app/src/main/java/io/ipoli/android/app/ui/dialogs/AndroidCalendarsPickerFragment.java (hunk 1 above)
Comment (polyvjk) [inline app/src/main/java/io/ipoli/android/app/ui/dialogs/AndroidCalendarsPickerFragment.java]: Forgot this one ;)

**Comment by curiousily** [comment]
This should address #57

**Pull request merged**